File: leibniz/pascal/leibniz.pas
program Leibniz;

function powersign(n: int64): int64;
begin
    if (n mod 2 = 0) then
        powersign := 1
    else
        powersign := -1;
end;

function leibniz(n: int64): double;
var
    s: double;
    i: int64;
begin
    s := 0;
    for i := 0 to n do s += powersign(i) / (2 * i + 1);
    leibniz := 4 * s;
end;

var
    n, code: int64;
begin
    val(paramstr(1), n, code);
    writeln(leibniz(n));
end.

File: leibniz/julia/leibniz.jl
function powersign(n)
    if n % 2 == 0
        1
    else
        -1
    end
end

function leibniz(n)
    s = 0.
    for k in 0:n
        s += powersign(k) / (2k + 1)
    end
    4s
end

const n = tryparse(Int64, ARGS[1])
println(leibniz(n))
